src/test/scala/mesosphere/mesos/ResourceMatcherTest.scala

```diff
@@ -253,8 +252,8 @@ class ResourceMatcherTest extends MarathonSpec with Matchers with Inside {
       )
     )
 
-    res.portsMatch.hostPortsWithRole.toSet should be(
-      Set(Some(PortsMatcher.PortWithRole(ResourceRole.Unreserved, 80, reservation = Some(portsReservation))))
+    res.portsMatch.get.hostPortsWithRole.toSet should be(
```

Comment:
split this into separate tests for res.portsMatch.nonEmpty and res.portsMatch.foreach { _.hostPortsWithRole.toSet should be ... }